lib/s3routes/routesUtils.ts

```diff
@@ -22,20 +22,33 @@ export type CallApiMethod = (
     callback: (err: ArsenalError | Error | null, ...data: any[]) => void,
 ) => void;
 
+type APIError = {
+    // Some VaultClient errors have the type of error in the "code" property
+    code?: string | number;
+    // Other external API errors may have the type of error in "message" (this is the case
+    // for Metadata API errors for example)
+    message?: string;
+    description?: string;
+}
+
 /**
  * Convert an error to an ArsenalError instance
  * @param err - Error to convert
  * @returns ArsenalError instance
  */
-export function toArsenalError(err: ArsenalError | Error): ArsenalError {
+export function toArsenalError(err: ArsenalError | APIError ): ArsenalError {
     if (err instanceof ArsenalError) {
         return err;
     }
-
     // If it's a known error type, use the error instance, otherwise wrap
     // it in an InternalError
-    return errorInstances[err.message] ||
-        errorInstances.InternalError.customizeDescription(err.message);
+    return err.message && errorInstances[err.message] ||
+        err.code && errorInstances[err.code] ||
```

Comment:
do we really need to check if the field is present? indexing an object with value `undefined` just returns undefined (`{ foo:1, bar:2 }[undefined] === undefined`), so we may just do the following ?
```suggestion
    return errorInstances[err.message] ||
        errorInstances[err.code] ||
```

Reply:
or even simpler, maybe this:
```suggestion
    return errorInstances[err.message || err.code] ||
```

(unless we have situations where both `message` and `code` are set, but message does not match so we need to rely on code for matching arsenal error...)

Reply:
Because of the different non overlapping error types, this doesn't build in TS.
The only "clutter free" way i see would be doing something like : `errorInstances[(err as any)?.code]`

Reply:
would this work?
```
return errorInstances[err.message! || err.code!] ||
```

Reply:
This worked `errorInstances[(err as Error).message || (err as VaultError).code] `